src/libcore/result.rs

```diff
@@ -909,6 +909,38 @@ impl<T: Default, E> Result<T, E> {
     }
 }
 
+#[unstable(feature = "inner_deref", reason = "newly added", issue = "50264")]
+impl<T: Deref, E: Deref> Result<T, E> {
```

Comment:
For maximum usability, these should probably be three impl blocks. One for T: Deref, one for E: Deref and one with both bounds. Otherwise you can't use `deref_ok` on a `Result<String, u32>` I think.